test/desdemona/tasks/kafka_test.clj

```diff
@@ -1,6 +1,35 @@
 (ns desdemona.tasks.kafka-test
   (:require [clojure.test :refer [deftest is]]
-            [desdemona.tasks.kafka :refer [deserialize-message-raw]]))
+            [clojure.java.io :as io]
+            [cheshire.core :as json]
+            [desdemona.tasks.kafka :refer [deserialize-message-raw deserialize-message-json]]))
+
+(deftest deserialize-message-json-test
+  (let [in {"this" "will"
+            "BE" "json"
+            "WITH_DIFFERENT" "forms"
+            "_that_we" "will change"}
+        encoded (json/generate-string in)
+        got (deserialize-message-json (.getBytes encoded))]
+    (is (= got {:this "will"
+                :be "json"
+                :with-different "forms"
+                :that-we "will change"}))))
```

Comment:
Nitpick: somehow clojure.test  ended up picking Yoda assertions, so it thinks (for equality) that the _first_ one is the expected value and the second is the actual value. This makes sense for the multi-arity = case, since if you want to want to check against the same expected value, it probably makes more sense to do `(= expected first-way-to-get-there second-way-to-get-there)` than `(= first-way expected second-way)`.

This only matters for test failure reporting: it will get "expected" and "actual" backwards.